libraries/AP_Scripting/examples/plane-doublets.lua

```diff
@@ -0,0 +1,159 @@
+-- This script will preform a control surface doublet
+-- The desired axis to perform a doublet on is selected and started with a momentary switch
+-- The plane maintains trim conditions while the switch is activated using servo overrides
+-- If the momentary switch is released before "DOUBLET FINISHED" is seen on the GCS,
+-- the aircraft recovers to FBWA mode to assist the pilot.
+-- Magnitude and duration of the doublet can also be controlled.
+-- It is suggested to allow the aircraft to trim for straight, level, unaccelerated flight (SLUF) in FBWB mode before
+-- starting a doublet
+-- Charlie Johnson, Oklahoma State University 2020
+
+local DOUBLET_ACTION_CHANNEL = 6 -- RCIN channel to start a doublet when high (>1700)
+local DOUBLET_CHOICE_CHANNEL = 7 -- RCIN channel to choose elevator (low) or rudder (high)
+local DOUBLET_FUCNTION = 19 -- which control surface (SERVOx_FUNCTION) number will have a doublet happen
+-- A (Servo 1, Function 4), E (Servo 2, Function 19), and R (Servo 4, Function 21)
+local DOUBLET_MAGNITUDE = 6 -- defined out of 45 deg used for set_output_scaled
+local DOUBLET_TIME = 500 -- period of doublet signal in ms
+
+-- flight mode numbers for plane https://mavlink.io/en/messages/ardupilotmega.html
+local MODE_MANUAL = 0
+local MODE_FBWA = 5
+local MODE_FBWB = 6
+local MODE_RTL = 11
+local K_AILERON = 4
+local K_ELEVATOR = 19
+local K_THROTTLE = 70
+local K_RUDDER = 21
+
+-- store timing information during doublet
+local start_time = -1
+local end_time = -1
+local now = -1
+local desired_mode = -1
```

Comment:
Variable "desired_mode" not used in any logical operation. Just set between current mode and -1? Is there a larger purpose for this variable or can it be removed?

Reply:
I added that as an intention for the retry_set_mode, but I didn't finish that because it was working as intended without it.